blockservice/blockservice.go

```diff
@@ -8,20 +8,33 @@ import (
 	"fmt"
 
 	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
-	process "github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess"
-	procrl "github.com/jbenet/go-ipfs/Godeps/_workspace/src/github.com/jbenet/goprocess/ratelimit"
 	blocks "github.com/jbenet/go-ipfs/blocks"
 	"github.com/jbenet/go-ipfs/blocks/blockstore"
+	worker "github.com/jbenet/go-ipfs/blockservice/worker"
 	exchange "github.com/jbenet/go-ipfs/exchange"
 	u "github.com/jbenet/go-ipfs/util"
 )
 
+var wc = worker.Config{
+	// When running on a single core, NumWorkers has a harsh negative effect on
+	// throughput. (-80% when < 25)
+	// Running a lot more workers appears to have very little effect on both
+	// single and multicore configurations.
+	NumWorkers: 25,
+
+	// These have no effect on when running on multiple cores, but harsh
+	// negative effect on throughput when running on a single core
+	// On multicore configurations these buffers have little effect on
+	// throughput.
+	// On single core configurations, larger buffers have severe adverse
+	// effects on throughput.
+	ClientBufferSize: 0,
+	WorkerBufferSize: 0,
```

Comment:
Right now, the client worker is always ready to receive from the client. We may need to reconsider the buffer if the worker starts doing non-trivial processing on the queue.